src/components/SearchBar/SearchBar.tsx

```diff
@@ -0,0 +1,73 @@
+import React, { type FC } from 'react';
+
+import searchIcon from './assets/search-icon.svg';
+import { DEFAULT_PLACEHOLDER } from './SearchBar.constants';
+import { useSearchBar } from './SearchBar.hook';
+import { SearchContainer, StyledInput, SearchIconButton } from './SearchBar.styles';
+
+import type { SearchBarProps } from './SearchBar.types';
+
+export const SearchBar: FC<SearchBarProps> = (props) => {
+  const {
+    placeholder = DEFAULT_PLACEHOLDER,
+    inputOnChange,
+    onSearch,
+    debounceDelay,
+    value,
+    defaultValue,
+    onChange,
+    disabled,
+    ...otherProps
+  } = props;
+
+  const hookProps = {
+    onSearch,
+    debounceDelay,
+    value,
+    defaultValue,
+    onChange,
+  };
+
+  const {
+    value: inputValue,
+    handleInputChange,
+    handleKeyDown,
+    handleButtonClick,
+  } = useSearchBar(hookProps);
+
+  const combinedHandleChange = (e: React.ChangeEvent<HTMLInputElement | HTMLTextAreaElement>) => {
+    // Type assertion para HTMLInputElement
+    const inputEvent = e as React.ChangeEvent<HTMLInputElement>;
+    handleInputChange(inputEvent);
+
+    if (inputOnChange) {
+      inputOnChange(e);
+    }
+  };
+
+  return (
+    <SearchContainer data-testid="search-bar-container">
+      <SearchIconButton
+        onClick={handleButtonClick}
+        aria-label="search"
```

Comment:
The button has aria-label="search" while the input uses aria-label "Search content". Accessible names should be consistent and preferably receive their strings from props/i18n so they can be localized. Also consider passing the aria-label as a prop so consumers can customize it.

**Sugerencia:**
```typescript
Use a single label value (from props or i18n) for both button and input where appropriate, e.g. aria-label={ariaLabel ?? t('search')}. Ensure casing/phrasing matches to prevent confusing screen reader output.
```